File: 4 - JAVA/4 - Matriz - ler-matriz-quadrada-dar-soma-positivos-escolhe-linha-coluna-mostra-diag-prin-e-altera.java
import java.util.Locale;
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {

		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);

		int N, linha, coluna;
		double soma;

		System.out.print("Qual a ordem da matriz? ");
		N = sc.nextInt();

		double[][] mat = new double[N][N];

		for (int i=0; i<N; i++) {			
			for (int j=0; j<N ; j++) {	
				System.out.printf("Elemento [%d,%d]: ", i, j);
				mat[i][j] = sc.nextDouble();
			}
		}

		System.out.println();
		soma=0;
		for (int i=0; i<N; i++) {		     		
			for (int j=0; j<N ; j++) {	
				if (mat[i][j] > 0) {
					soma = soma + mat[i][j];
				}
			}  
		}
		System.out.printf("\nSOMA DOS POSITIVOS: %.1f\n\n", soma);

		System.out.print("Escolha uma linha");
		linha = sc.nextInt();

		System.out.print("LINHA ESCOLHIDA: ");
		for (int j=0; j<N; j++) {
			System.out.printf("%.1f ", mat[linha][j]);
		}

		System.out.printf("\n\nEscolha uma coluna: ");
		coluna = sc.nextInt();

		System.out.print("COLUNA ESCOLHIDA: ");
		for (int i=0; i<N; i++) {		     		
			System.out.printf("%.1f ", mat[i][coluna]);
		}

		System.out.printf("\n\nDIAGONAL PRINCIPAL: ");
		for (int i=0; i<N; i++) {
			System.out.print(mat[i][i]);
		}

		for (int i=0; i<N; i++) {		     		
			for (int j=0; j<N ; j++) {
				if (mat[i][j] < 0) {
					mat[i][j] = Math.pow(mat[i][j], 2);
				}
			}
		}

		System.out.println("\n\nMATRIZ ALTERADA");
		for (int i=0; i<N; i++) {
			for (int j=0; j<N; j++) {
				System.out.printf("%.1f ", mat[i][j]);
			}
			System.out.println();
		}


		sc.close();
	}
}

File: 3 - Linguagem C++/4 - Matriz - ler-matriz-quadrada-dar-soma-positivos-escolhe-linha-coluna-mostra-diag-prin-e-altera.cpp
#include <iostream>
#include <iomanip>
#include <cmath>

using namespace std;

int main(){

    int N, somaPosit = 0, linha, coluna;

    cout << "Qual a ordem da matriz? ";
    cin >> N;

    double mat[N][N];

    for (int i=0; i<N; i++) {
      for (int j=0; j<N ; j++) {
        cout << "Elemento [" << i << "," << j << "]: ";
        cin >> mat[i][j];
       }
    }

    for (int i=0; i<N; i++) {
        for (int j=0; j<N; j++) {
             if (mat[i][j] > 0){
                somaPosit = somaPosit + mat[i][j];
             }
        }
    }

    cout << fixed << setprecision(1);
    cout << "SOMA DOS POSITIVOS: " << somaPosit << endl;

    cout << "Escolha uma linha: " << endl;
    cin >> linha;

    cout << "LINHA ESCOLHIDA: ";
    for (int i=0; i<N; i++) {
        cout << mat[linha][i] << " ";
     }

    cout << endl << endl << "Escolha uma coluna: " << endl;
    cin >> coluna;

    cout << endl << "COLUNA ESCOLHIDA: ";
    for (int i=0; i<N; i++) {
         cout << mat[i][coluna] << " ";
     }

     cout << endl << endl << "DIAGONAL PRINCIPAL: ";
     for (int i=0; i<N; i++) {
         cout << mat[i][i] << " ";
      }

     for (int i=0; i<N; i++) {
        for (int j=0; j<N; j++) {
            if (mat[i][j] < 0) {
              mat[i][j] = pow(mat[i][j],2);
           }
        }
     }

     cout << endl << endl << "MATRIZ ALTERADA: " << endl;
     for(int i=0; i<N; i++){
        for(int j=0; j<N; j++){
           cout << mat[i][j] << " ";
        }
        cout << endl;
    }

    return 0;
}
